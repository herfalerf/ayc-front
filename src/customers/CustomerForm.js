import React from "react";
import { useHistory } from "react-router-dom";
import { Formik, Form, Field } from "formik";
import { TextField } from "formik-material-ui";
import { Button } from "@material-ui/core";
import * as yup from "yup";

// Customer Form.
//
// Shows add customer form
// On submission calls addCustomer function prop
// redirects to /admin/customers
//
// Routes -> CustomerManager -> CustomerForm
// Routed at /admin/customer/add
//

const CustomerForm = ({ addCustomer }) => {
  const phoneRegExp =
    /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;

  const schema = yup.object().shape({
    name: yup.string().required("Name required"),
    email: yup
      .string()
      .email("Invalid email format")
      .required("Email Required"),
    phone: yup
      .string()
      .matches(
        phoneRegExp,
        "Please enter phone number in format ###-###-#### or ##########"
      ),
  });
  const history = useHistory();
  console.debug("ContactForm", "addCustomer=", typeof addCustomer);

  return (
    <div>
      <Formik
        validationSchema={schema}
        initialValues={{
          name: "",
          email: "",
          company: "",
          phone: "",
        }}
        onSubmit={async (values, { setStatus, setSubmitting }) => {
          setStatus(undefined);
          let result = await addCustomer(values);

          if (result.success) {
            setSubmitting(false);
            history.push("/admin/customers");
          } else {
            if (result.errors[0] === `Email already exists: ${values.email}`) {
              setSubmitting(false);
              history.push("/");
            } else {
              setStatus({ error: result.errors });
            }
          }
        }}
      >
        {({ isSubmitting, status }) => (
          <Form>
            <Field
              label="name*"
              type="name"
              name="name"
              component={TextField}
            />
            <br />
            <Field
              label="email*"
              type="email"
              name="email"
              component={TextField}
            />
            <br />
            <Field
              label="company"
              type="company"
              name="company"
              component={TextField}
            />
            <br />
            <Field
              label="phone"
              type="phone"
              name="phone"
              component={TextField}
            />
            <br />
            {status && status.error ? (
              <div>API Error: {status.error}</div>
            ) : null}
            <br />
            <Button
              variant="contained"
              color="primary"
              disabled={isSubmitting}
              type="submit"
            >
              Submit
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  );
};

export default CustomerForm;
