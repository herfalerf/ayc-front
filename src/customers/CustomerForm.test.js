import React from "react";
import { render } from "@testing-library/react";
import CustomerForm from "./CustomerForm";

it("renders without crashing", function () {
  render(<CustomerForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<CustomerForm />);
  expect(asFragment()).toMatchSnapshot();
});
