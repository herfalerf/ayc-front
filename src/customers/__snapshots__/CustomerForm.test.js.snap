// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-upper-4"
    />
    <div
      class="MuiGrid-root makeStyles-form-5 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root makeStyles-message-6 MuiTypography-h5"
        >
          Add a New Customer
        </h5>
        <form
          action="#"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              name*
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="name"
                type="name"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              email*
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              company
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="company"
                type="company"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              phone
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="phone"
                type="phone"
                value=""
              />
            </div>
          </div>
          <br />
          <br />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
